internal/tiltfile/k8s.go

```diff
@@ -382,43 +382,52 @@ func (s *tiltfileState) k8sResourceV2(thread *starlark.Thread, fn *starlark.Buil
 		return nil, err
 	}
 
+	resourceName := workload
+	needsToHaveObjects := false
 	if workload == "" {
-		return nil, fmt.Errorf("%s: workload must not be empty", fn.Name())
+		resourceName = newName
+		// If a resource doesn't specify a workload then it needs to have objects to be valid
+		needsToHaveObjects = true
```

Comment:
nbd but why not check here whether they passed the objects param, rather than storing a bool and waiting til the bottom of the function?

Reply:
It felt weird to me to parse the parameters out of order, but maybe that's not a big deal.